### PR TITLE
Add link to proposal: Asynchronous JSON methods

### PR DESCRIPTION
**Pull request by mohsen1**

### CODE REVIEW

**Comment by zloirock** [comment]
Your polyfill resolves async, but performs sync
```js
return new Promise(resolve => resolve(JSON.stringify(object)));
```
Use something like
```js
return Promise.resolve().then(_ => JSON.parse(object));
```

**Comment by mohsen1** [comment]
AFAIK no matter what you do, parse will happen sync.

**Comment by zloirock** [comment]
It will happen in main thread, but, in second example, async.

**Comment by getify** [comment]
@mohsen1 the solution @zloirock suggests moves the **synchronous** parse to the next event loop tick, which has (at least) the effect of unblocking the current event loop tick, which is an approximation for what we all call "async" (from the calling code's perspective).

@zloirock the shame of that is that there's an extra Task fired (two instead of one) to get the end result in the calling code. Also, as I [suggest here](https://github.com/mohsen1/async-json/issues/5), if the polyfill changes to work as you suggest, it's setting more of a precedent that an implementation cannot do a synchronous parse (so one Task instead of two+) even if it feels that would be more appropriate. I think that's a bad idea.

**Comment by zloirock** [comment]
@getify i think, bad idea use "polyfill" for async features which blocks current sync flow. I don't see any problem in extra microtasks for async feature. ES6 Promise always async, your issue rather is a problem of implementation.

**Comment by mohsen1** [comment]
@getify thanks for clearing that up. I wasn't aware that you can perform operations in the next tick with that syntax.

@zloirock I've wrote about how to abuse `Response#json` from fetch API to perform parse in a background thread [here](http://azimi.me/2015/07/30/non-blocking-async-json-parse.html). Only chrome does `.json` in a non-blocking manner but if other browsers adopt it maybe it's worth putting it in a polyfill.  It's a hack but hey...

**Comment by vkurchatkin** [comment]
@mohsen1 someone posted link to your article about `Request.json` on io.js issue tracker and I couldn't help but object: https://github.com/nodejs/io.js/issues/2031#issuecomment-126840751

TLDR there is no way this could work (and it doesn't). It's nothing more than `return Promise.resolve().then(_ => JSON.parse(object));`

**Comment by mohsen1** [comment]
From what I learned from https://github.com/nodejs/io.js/issues/2031implementation of this is very hard if not impossible. There is also thread safety issues https://github.com/mohsen1/async-json/issues/6

It's like a flying car, the idea is great but implementation is really hard...

**Comment by thomasfoster96** [comment]
Unless I'm missing something... this proposal doesn't otherwise make much sense to me, because it's either moving `JSON.parse` to the next event loop tick (which can be done right now anyway), or making a single function in the language somewhat concurrent.

If the issue here is something blocking the main thread, wouldn't it be better to think about adding something to the language for easily moving things to a background thread? Maybe something similar to goroutines or moving Web Workers into ES?
```javascript
(go JSON.parse(lotsOfJSON)).then(doSomething);
```

**Comment by benjamingr** [comment]
The problem with generic "moving to a background thread" is serialization of input and output. You need a way to bypass that process when your goal is making serialiIng itself faster.

I do agree that a solution that does not take generic return value serialization or lack of into account is problematic though

**Comment by thomasfoster96** [comment]
@benjamingr yeah I should have emphasised the 'easily' is referring to using the feature, actually making it work well would take a lot of work.

**Comment by bterlson** [comment]
Closing. All stage zero proposals must have an identified TC39 champion. See CONTRIBUTING.md for more details!
